src/magma/vk/commandBuffer.cpp

```diff
@@ -43,6 +43,28 @@ void CommandBufferArr::endCmdBuf(uint32_t i)
     VK_CHECK_ERR(result, "failed to record command buffer!");
 }
 
+void CommandBufferArr::endAndSubmitCmdBuf(uint32_t i, VkQueue queue) {
+    VkResult result = vkEndCommandBuffer(commandBuffers[i]);
+    VK_CHECK_ERR(result, "failed to record command buffer!");
+
+    VkSubmitInfo submitInfo{};
+    submitInfo.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+    submitInfo.commandBufferCount = 1;
+    submitInfo.pCommandBuffers = &commandBuffers[i];
+
+    vkQueueSubmit(queue, 1, &submitInfo, VK_NULL_HANDLE);
+    vkQueueWaitIdle(queue);
```

Comment:
Do we really need vkQueueWaitIdle all the time when we submit cmd buffer?

Reply:
Without idle we will have this problem
![image](https://user-images.githubusercontent.com/32343654/108279474-b66f3180-718d-11eb-878c-86c4235afdc5.png)

Reply:
This is because it is used in places where before were SingleTimeCommandBuffer, which submit was sync.

By the way, I think, that such sync methods should be marked by "Sync" or "_sync" suffix.